hyperspy/tests/test_io.py

```diff
@@ -189,120 +188,110 @@ def test_file_not_found_error():
 
-        t = hs.load(Path(dirpath, "temp.hspy"))
-        assert t.original_metadata.as_dictionary() == s.original_metadata.as_dictionary()
+    t = hs.load(tmp_path / "temp.hspy")
+    assert t.original_metadata.as_dictionary() == s.original_metadata.as_dictionary()
 
-        t = hs.load(Path(dirpath, "temp.hspy"), load_original_metadata=False)
-        assert t.original_metadata.as_dictionary() == {}
+    t = hs.load(tmp_path / "temp.hspy", load_original_metadata=False)
+    assert t.original_metadata.as_dictionary() == {}
 
 
-def test_load_save_filereader_metadata():
+def test_load_save_filereader_metadata(tmp_path):
     # tests that original FileReader metadata is correctly persisted and
     # appended through a save and load cycle
 
-    my_path = os.path.dirname(__file__)
-    s = hs.load(os.path.join(my_path, "msa_files", "example1.msa"))
+    s = hs.datasets.example_signals.EDS_TEM_Spectrum()
     assert s.metadata.General.FileIO.Number_0.io_plugin == \
-           'rsciio.msa.api'
+           'rsciio.hspy.api'
     assert s.metadata.General.FileIO.Number_0.operation == 'load'
     assert s.metadata.General.FileIO.Number_0.hyperspy_version == hs_version
 
-    with tempfile.TemporaryDirectory() as dirpath:
-        f = os.path.join(dirpath, "temp")
-        s.save(f)
-        expected = {
-            '0': {
-                'io_plugin': 'rsciio.msa.api',
-                'operation': 'load',
-                'hyperspy_version': hs_version
-            },
-            '1': {
-                'io_plugin': 'rsciio.hspy.api',
-                'operation': 'save',
-                'hyperspy_version': hs_version
-            },
-            '2': {
-                'io_plugin': 'rsciio.hspy.api',
-                'operation': 'load',
-                'hyperspy_version': hs_version
-            },
-        }
-        del s.metadata.General.FileIO.Number_0.timestamp  # runtime dependent
-        del s.metadata.General.FileIO.Number_1.timestamp  # runtime dependent
-        assert \
-            s.metadata.General.FileIO.Number_0.as_dictionary() == expected['0']
-        assert \
-            s.metadata.General.FileIO.Number_1.as_dictionary() == expected['1']
-
-        t = hs.load(Path(dirpath, "temp.hspy"))
-        del t.metadata.General.FileIO.Number_0.timestamp  # runtime dependent
-        del t.metadata.General.FileIO.Number_1.timestamp  # runtime dependent
-        del t.metadata.General.FileIO.Number_2.timestamp  # runtime dependent
-        assert t.metadata.General.FileIO.as_dictionary() == expected
+    f = tmp_path / "temp"
+    s.save(f)
+    expected = {
+        '0': {
+            'io_plugin': 'rsciio.hspy.api',
+            'operation': 'load',
+            'hyperspy_version': hs_version
+        },
+        '1': {
+            'io_plugin': 'rsciio.hspy.api',
+            'operation': 'save',
+            'hyperspy_version': hs_version
+        },
+        '2': {
+            'io_plugin': 'rsciio.hspy.api',
```

Comment:
This now duplicates the item `'0'` above. Is it just to make sure that the saved file can be reread?

Reply:
They have a time difference, but in the test it is removed to simplify the test.